### PR TITLE
Create db schema

### PR DESCRIPTION
**Pull request by pkinglinz**
Table with comments to make automatic creation of table using read the docs easy.

refers to linz/qgis-buildingsgeneralisation-plugin#250

Please look over structure especially comments and default values

### CODE REVIEW

File: db/sql/schema_and_tables.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    , internal boolean NOT NULL DEFAULT false
+    , internal_comments character varying(100) DEFAULT ''
+    , shape public.geometry(MultiPolygon, 2193) NOT NULL
+);
+
+CREATE INDEX shx_nz_facilities
+    ON facilities_lds.nz_facilities USING gist (shape);
+
+COMMENT ON TABLE facilities_lds.nz_facilities IS
+'The nz_facilities table holds geometries originating from NationalMap facility polygons '
```
Comment (jducnuigeen) [inline db/sql/schema_and_tables.sql]: Maybe leave the mention of orginating from NationalMap to the metadata?

File: db/sql/schema_and_tables.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+COMMENT ON COLUMN facilities_lds.nz_facilities.facility_id IS
+'The unique identifier for each geometry.';
+COMMENT ON COLUMN facilities_lds.nz_facilities.external_facility_id IS
+'The unique identifier of this facility used by the authoritative source';
+COMMENT ON COLUMN facilities_lds.nz_facilities.name IS
+'The name of the facility.';
+COMMENT ON COLUMN facilities_lds.nz_facilities.external_name IS
+'The name of the facility used by the authoritative source.';
+COMMENT ON COLUMN facilities_lds.nz_facilities.use IS
+'The generic use of the facility.';
```
Comment (jducnuigeen) [inline db/sql/schema_and_tables.sql]: Do we want to mention the generic use matches those uses from Topo50?

File: db/sql/schema_and_tables.sql [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+COMMENT ON COLUMN facilities_lds.nz_facilities.use_type IS
+'Use type as defined by the authoritative source.';
+COMMENT ON COLUMN facilities_lds.nz_facilities.use_subtype IS
+'Use subtype as defined by the authoritative source.';
+COMMENT ON COLUMN facilities_lds.nz_facilities.estimated_occupancy IS
+'An approximation of the occupancy from the authoritative source. It may not include staff.';
+COMMENT ON COLUMN facilities_lds.nz_facilities.last_modified IS
+'The most recent date on which any attribute or geometry that is part of the facility was modified.';
+COMMENT ON COLUMN facilities_lds.nz_facilities.internal IS
+'Identifies features which will not be added to the LDS, such as teen parenting units which are part of a school.';
```
Comment (jducnuigeen) [inline db/sql/schema_and_tables.sql]: I would leave out the example of teen parenting in this comment field, and put this kind of example or something similar into the metadata.

File: db/sql/schema_and_tables.sql [hunk 4]
```diff
@@ -0,0 +1,56 @@
+--Schema
+
+CREATE SCHEMA IF NOT EXISTS facilities_lds;
```
Comment (Douglas-K) [inline db/sql/schema_and_tables.sql]: To be consistent with buildings, there should be 2 schemas and 2 tables.
- `facilities` schema should have a `facilities` table including internal columns
- `facilities_lds` schema should have a `nz_facilities` table with only the columns to be published on the LDS

**Review by Douglas-K — CHANGES_REQUESTED** [review]
Looks pretty good, except that it should be 2 schemas and 2 tables.  See in-line comment.

File: db/sql/schema_and_tables.sql (hunk 2 above)
Comment (Douglas-K) [reply db/sql/schema_and_tables.sql]: I wouldn't mention Topo50, as people might get confused and the link isn't that strong.  Also there are uses that are not in Topo50 like Supermarket and potentially Rest home.

File: db/sql/schema_and_tables.sql (hunk 3 above)
Comment (Douglas-K) [reply db/sql/schema_and_tables.sql]: Agree.  Maybe better examples would be 'such as schools that are closed or proposed.'

**Review by Douglas-K — CHANGES_REQUESTED** [review]
Looks pretty good, except that it should be 2 schemas and 2 tables.  See in-line comment, plus some other minor comments.

**Review by Douglas-K — APPROVED** [review]

**Review by jducnuigeen — APPROVED** [review]

**Pull request merged**